packages/pyright-internal/src/common/extensibility.ts

```diff
@@ -39,12 +39,22 @@ export interface ProgramExtension {
 // Readonly wrapper around a Program. Makes sure it doesn't mutate the program.
 export interface ProgramView {
     readonly id: number;
-    rootPath: string;
-    getImportResolver(): ImportResolver;
-    console: ConsoleInterface;
-    getConfigOptions(): ConfigOptions;
+    readonly rootPath: string;
```

Comment:
we should go through everything we return here and make sure we can't mutate them. for example, `configOptions` should have all field field marked as `readonly`. also, we might want to create `ImportResolverView` or something that only expose `readonly` APIs that LSP feature actually uses. so, blocking IDE features from calling things like `invalidateCache` and etc. We probably want to remove `Completion` code in `ImportResolver` as well.

currently we expose `SourceFileInfo` through view, we should either not expose it or expose only `readonly` fields.